drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,342 @@
+			     int16_t *valp)
+{
+	const struct max17262_config *cfg = dev->config;
+	uint8_t i2c_data[2];
+	int rc;
+
+	rc = i2c_burst_read(cfg->i2c, cfg->i2c_addr, reg_addr,
+			    i2c_data, 2);
+	if (rc < 0) {
+		LOG_ERR("Unable to read register");
+		return rc;
+	}
+	*valp = ((int16_t)i2c_data[1] << 8) | i2c_data[0];
+
+	return 0;
+}
+
+/**
+ * @brief Write a register value
+ *
+ * Registers have an address and a 16-bit value
+ *
+ * @param dev MAX17262 device to access
+ * @param reg_addr Register address to write to
+ * @param val Register value to write
+ * @return 0 if successful, or negative error code from I2C API
+ */
+static int max17262_reg_write(const struct device *dev, uint8_t reg_addr,
+			     int16_t val)
+{
+	const struct max17262_config *cfg = dev->config;
+	uint8_t i2c_data[3] = {reg_addr, val & 0xFF, (uint16_t)val >> 8};
+
+	return i2c_write(cfg->i2c, i2c_data, sizeof(i2c_data),
+		     cfg->i2c_addr);
+}
+
+/**
+ * @brief Convert sensor value from millis
+ *
+ * @param val Where to store converted value in sensor_value format
+ * @param val_millis Value in millis
+ */
+static void convert_millis(struct sensor_value *val, int32_t val_millis)
+{
+	val->val1 = val_millis / 1000;
+	val->val2 = (val_millis % 1000) * 1000;
+}
+
+/**
+ * @brief Convert raw register values for specific channel
+ *
+ * @param dev MAX17262 device to access
+ * @param chan Channel number to read
+ * @param valp Returns the sensor value read on success
+ * @return 0 if successful
+ * @return -ENOTSUP for unsupported channels
+ */
+static int max17262_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *valp)
+{
+	const struct max17262_config *const config = dev->config;
+	struct max17262_data *const data = dev->data;
+	int32_t tmp;
+
+	switch (chan) {
+	case SENSOR_CHAN_GAUGE_VOLTAGE:
+		/* Get voltage in uV */
+		tmp = data->voltage * 1250 / 16;
+		/* Convert to V */
+		valp->val1 = tmp / 1000000;
+		valp->val2 = tmp % 1000000;
+		break;
+	case SENSOR_CHAN_GAUGE_AVG_CURRENT: {
+		int current;
+		/* Get avg current in nA */
+		current = data->avg_current * 156250;
+		/* Convert to mA */
+		valp->val1 = current / 1000000;
+		valp->val2 = current % 1000000;
+		break;
+	}
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		valp->val1 = data->state_of_charge / 256;
+		valp->val2 = data->state_of_charge % 256 * 1000000 / 256;
+		break;
+	case SENSOR_CHAN_GAUGE_TEMP:
+		valp->val1 = data->internal_temp / 256;
+		valp->val2 = data->internal_temp % 256 * 1000000 / 256;
+		break;
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		convert_millis(valp, data->full_cap);
+		break;
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		convert_millis(valp, data->remaining_cap);
+		break;
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		/* Get time in ms */
+		if (data->time_to_empty == 0xffff) {
+			valp->val1 = 0;
+			valp->val2 = 0;
+		} else {
+			tmp = data->time_to_empty * 5625;
```

Comment:
Where does this constant come from? This and a few other non-obvious constants used in this driver could benefit from being defined as preprocessor macros.

Reply:
From the following table in the MAX1726x ModelGauge m5 EZ User Guide:
https://pdfserv.maximintegrated.com/en/an/user-guide-6597-max1726x-m5-ez-rev3-p4.pdf#%5B%7B%22num%22%3A53%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C70%2C302%2C0%5D

Reply:
Added a few defines to mask that magic numbers. Please advise if you would like to implement some more.